### ISSUE TITLE
HLSL modf out-parameter not converting properly

### ISSUE THREAD

**Issue by redorav** [KhronosGroup/glslang#773]
Modf is a fairly unusual function, but we've found a case in our code where it doesn't compile.

```
float4 test_px(float4 input : TEXCOORD0)
{
    float4 intPart;
    float3 foo = modf(input.xyz, intPart.xyz);
    return foo.xyzx;
}
```

If I make intPart a float3 it compiles correctly. And for example sincos (which also takes out variables) works fine under this example. It's an easy fix on our side but just making you aware. Thanks!

**Comment by johnkslang** [comment]
I think this is the in/out conversion issue.

**Comment by johnkslang** [comment]
This was also fixed by bbbd9a2a1f41ab301861c6052d9643c626ee9961, fixing #1843.

It is not about normal argument conversion, but specific to built-in function output parameters.
